packages/headless/src/features/search/search-actions-thunk-processor.ts

```diff
@@ -263,6 +257,55 @@ export class AsyncSearchThunkProcessor<RejectionType> {
     };
   }
 
+  private async processClassicDidYouMeanAutoCorrection(
+    originalSearchSuccessResponse: SearchResponseSuccess
+  ): Promise<ExecuteSearchThunkReturn | RejectionType> {
+    const originalQuery = this.getCurrentQuery();
+
+    const {correctedQuery} = originalSearchSuccessResponse.queryCorrections[0];
+
+    const retried = await this.automaticallyRetryQueryWithCorrection(
+      correctedQuery
+    );
+
+    if (isErrorResponse(retried.response)) {
+      this.dispatch(logQueryError(retried.response.error));
+      return this.rejectWithValue(retried.response.error) as RejectionType;
+    }
+
+    return {
+      ...retried,
+      response: {
+        ...retried.response.success,
+        queryCorrections: originalSearchSuccessResponse.queryCorrections,
+      },
+      automaticallyCorrected: true,
+      originalQuery,
+      analyticsAction: logDidYouMeanAutomatic(),
+    };
+  }
+
+  private processFallbackQueryResultsDidYouMeanAutoCorrection(
+    originalFetchedResponse: FetchedResponse
+  ): ExecuteSearchThunkReturn {
```

Comment:
No promise reject for this one?

Reply:
No, in this code path we are not retrying any query, it's already been corrected at the API level and we have the proper results already. We just need to adjust client side state to what the query has been corrected to.

For the classic did you mean feature, client side code is in charge of doing the retry with the correction (which means there's an async request + request failure possibility)